lib/resource_methods.rb

```diff
@@ -155,7 +154,12 @@ def generate_uuid
 
     def create_with_id(id, **)
       obj = new(**)
-      obj.id = id
+      obj.id = case id
+      when String, nil
+        id
+      else
+        id.id
+      end
```

Comment:
maybe
```suggestion
      obj.id = if id.respond_to?(:id) 
        id.id
      else
        id
      end
```

Reply:
The `id = id.id if id.is_a?(Sequel::Model)` approach unfortunately breaks if you are using `instance_double`, and we are. So I decided against that.